Libraries/LibWeb/Painting/Paintable.cpp

```diff
@@ -158,6 +160,44 @@ StackingContext* Paintable::enclosing_stacking_context()
     VERIFY_NOT_REACHED();
 }
 
+void Paintable::paint_inspector_overlay(DisplayListRecordingContext& context) const
+{
+    Vector<Painting::Paintable const*> self_and_ancestors {};
+    for (Paintable const* paintable = this; paintable; paintable = paintable->parent()) {
+        self_and_ancestors.append(paintable);
+    }
+
+    for (auto const* paintable : self_and_ancestors.in_reverse()) {
+        if (auto const* box = as_if<PaintableBox>(paintable)) {
+            box->apply_scroll_offset(context);
+            if (box->stacking_context()) {
+                auto to_device_pixels_scale = float(context.device_pixels_per_css_pixel());
+                auto transform_matrix = box->transform();
+                auto transform_origin = box->transform_origin().to_type<float>();
+                // We only want the transform here, everything else undesirable for the inspector overlay
+                DisplayListRecorder::PushStackingContextParams push_stacking_context_params {
+                    .opacity = 1.0,
+                    .compositing_and_blending_operator = Gfx::CompositingAndBlendingOperator::Normal,
+                    .isolate = false,
+                    .transform = StackingContextTransform(transform_origin, transform_matrix, to_device_pixels_scale),
+                };
+                context.display_list_recorder().push_stacking_context(push_stacking_context_params);
+            }
+        }
+    }
+
+    paint_inspector_overlay_internal(context);
+
+    for (auto const* paintable : self_and_ancestors) {
+        if (auto const* box = as_if<PaintableBox>(paintable); box && box->stacking_context()) {
```

Comment:
You should remove `; box && box->stacking_context()` here since you check it again on the next line.